casadm/cas_lib.c

```diff
@@ -565,6 +591,154 @@ int get_dev_path(const char* disk, char* buf, size_t num)
 	return err;
 }
 
+/* Indicate whether given entry in /dev/disk/by-id should be ignored -
+   we ignore software created links like 'lvm-' since these can point to
+   both CAS exported object and core device depending on initialization order.
+*/
+static bool dev_link_blacklisted(const char* entry)
+{
+	static const char* const prefix_blacklist[] = {"lvm", "md-name"};
+	static const unsigned count = ARRAY_SIZE(prefix_blacklist);
+	const char* curr;
+	unsigned i;
+
+	for (i = 0; i < count; i++) {
+		curr = prefix_blacklist[i];
+		if (!strncmp(entry, curr, strnlen_s(curr, MAX_STR_LEN)))
+			return true;
+	}
+
+	return false;
+}
+
+/* Indicate whether given path should be passed without check */
+static bool is_dev_link_whitelisted(const char* path, size_t path_len)
+{
+	regex_t regex;
+	int result;
+	static const char* const whitelist[] = { "cas[:0-9:]\\+-[:0-9:]\\+" };
+	static const unsigned count = ARRAY_SIZE(whitelist);
+	char temp_path[MAX_STR_LEN];
+	size_t i;
+
+	if (strncpy_s(temp_path, MAX_STR_LEN, path, path_len))
+		return FAILURE;
+
+	for (i = 0; i < count; i++) {
+		result = regcomp(&regex, whitelist[i], 0);
```

Comment:
how about specifying REG_NOSUB? I think it makes sense, since you provide NULL pmatch to regexec anyway

Reply:
You're right